addons/pos_restaurant/static/src/app/split_bill_screen/split_bill_screen.js

```diff
@@ -52,13 +52,59 @@ export class SplitBillScreen extends Component {
         }
     }
 
+    _getLatestOrderNameStartingWith(name) {
+        const filter_orders = (order) => {
+            const orderName = order.getOrderName();
+            return orderName.startsWith(name) && orderName.length === name.length + 1;
+        };
+
+        const sort_by_last_char = (a, b) => {
+            const lastCharA = a[a.length - 1];
+            const lastCharB = b[b.length - 1];
+
+            if (lastCharA < lastCharB) {
+                return -1;
+            }
+            if (lastCharA > lastCharB) {
+                return 1;
+            }
+            return 0;
+        };
+
+        const orders = this.pos.get_open_orders().filter(filter_orders);
+
+        if (orders.length === 0) {
+            return name;
+        }
+
+        const orders_names = orders.map((order) => order.getOrderName());
+        const sorted_orders_names = orders_names.sort(sort_by_last_char);
+
+        return sorted_orders_names[sorted_orders_names.length - 1];
+    }
+
+    _getNewOrderName(originalOrderName) {
```

Comment:
shouldn't we have upper case letters in this case ? `6A`, `6B`, instead of `6a`, `6b`